addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -419,6 +419,12 @@ ListRenderer.include({
         this.defs = defs;
         _.each(this.columns, function (node, colIndex) {
             var $td = $tds.eq(colIndex);
+
+            // Restrict the list_activity widget as readonly
+            if (node.attrs.widget === "list_activity" && self.isInMultipleRecordEdition(recordID)) {
```

Comment:
activity is part of `mail` module at minimum, it should not be referenced it `web` (that's against our module/dependency hierarchy)

Reply:
@seb-odoo , 
Another way is to check for the generic one2many field and make it readonly 
like: https://github.com/odoo-dev/odoo/commit/983e8ec4acedc42057a86e0bf0802018ad10959f

Second way is to handle it from the list_Activity widget but it will add the parent dependency in widget list_activity which is not preferable.
like: https://github.com/odoo/odoo/commit/0b1bd9e79cfe2051304a3ee0ea7395471a2683ff

and last option is to split this method and override from mail. 

So I prefer to do generic for one2many field as one2many does not support multi_edit in list.

can you please provide your suggestion on it ? 

Thanks

Reply:
I agree that no widget (and in particular widgets from outside web/) should be referenced in the list renderer. I guess that you could add an attribute on AbstractField (e.g. `multi_edit`) that you can set to false to disable multi edition for specific widgets.

You don't need to do it for now, but this could as well be an attribute on `<field>` nodes in the arch, to disable multi edition in specific cases.

Reply:
@aab-odoo I improved this PR as per your suggestion, what do you think about this?

Reply:
It looks way more generic, so I think it's better. However, I don't like that you override the 'readonly' property in the attrs. Is it necessary?

Reply:
@aab-odoo I am going to improve it as it does not work properly as it does not apply o_readonly_modifier class table cell first time, Thanks